src/devices/Board/SpiDeviceManager.cs

```diff
@@ -0,0 +1,139 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Device.Spi;
+
+namespace Iot.Device.Board
+{
+    internal class SpiDeviceManager : SpiDevice
+    {
+        private readonly Board _board;
+        private readonly int[] _pins;
+        private SpiDevice _device;
+
+        public SpiDeviceManager(Board board, SpiConnectionSettings connectionSettings, int[]? pins, Func<SpiConnectionSettings, int[], SpiDevice> createOperation)
+        {
+            if (pins == null || pins.Length < 3)
+            {
+                throw new ArgumentException("Must provide a valid set of at least 3 pins", nameof(pins));
+            }
+
+            _board = board;
+            ConnectionSettings = connectionSettings;
+            try
+            {
+                _board.ReservePin(pins[0], PinUsage.Spi, this);
+                _board.ReservePin(pins[1], PinUsage.Spi, this);
+                _board.ReservePin(pins[2], PinUsage.Spi, this);
+                if (pins.Length == 4)
```

Comment:
as before you checked for pin lenght <3 and message about this, you should most likelt check for >=4. I agree, it doesn't change much anyway

Reply:
Now testing for exactly 3 or 4 pins.